api/tests/mocha/services/authorization.spec.js

```diff
@@ -1314,6 +1314,29 @@ describe('Authorization service', () => {
         });
     });
 
+    it('adds unassigned key if the user has required permissions', () => {
+      auth.hasAllPermissions.returns(true);
+      config.get.returns(true);
+
+      const docObjs = [
+        { // unallocated
+          doc: {
+            _id: 'unallocated', type: 'data_record',
+          },
+          viewResults: {
+            contactsByDepth: [],
+            replicationKeys: [['_unassigned', {}]]
+          },
+        },
+      ];
+
+      return service
+        .getScopedAuthorizationContext(userCtx, docObjs)
+        .then(result => {
+          result.subjectIds.should.have.members(['_all', '_unassigned', 'org.couchdb.user:user']);
```

Comment:
Consider adding a test, or extending an existing one, to assert that users without the permission do not have the '_unassigned' id. This is covered in the e2e test but it'd be nice to have unit coverage here too.

Reply:
Will do!